quickwit/quickwit-query/src/elastic_query_dsl/regex_query.rs

```diff
@@ -22,16 +22,22 @@ use crate::query_ast::{QueryAst, RegexQuery as AstRegexQuery};
 #[serde(deny_unknown_fields)]
 pub struct RegexQueryParams {
     value: String,
-    // we could probably add case_insensitive
+    #[serde(default)]
+    case_insensitive: bool,
 }
 
 pub type RegexQuery = OneFieldMap<RegexQueryParams>;
 
 impl ConvertibleToQueryAst for RegexQuery {
     fn convert_to_query_ast(self) -> anyhow::Result<QueryAst> {
+        let regex = if self.value.case_insensitive {
+            format!("(?i){}", self.value.value)
```

Comment:
til that syntax works in the flavour of regex we have

Reply:
I would have preferred to pass the mode outside of the regex as a flag to the regex constructor, but it seems that the underlying lib requires us to pass it as a modifier.